src/violation.rs

```diff
@@ -55,6 +55,8 @@ pub enum ViolationCode {
     TypeMatched,
     AllTypesNotMatched,
     TypeMismatched,
+    FieldsNotMatched,
```

Comment:
nitpick/non-blocking—I notices that we have some `ViolationCode`s that use "NotMatched" and others use "Mismatched". Doesn't need to be fixed now, but I think it would be a good idea to create an issue to go and review all of the public enum names and the violation messages for stylistic consistency before we release v1.0.0.

Reply:
I will create an issue for this clean up.